### ISSUE TITLE
Configuration as code

### ISSUE THREAD

**Issue by 12ian34** [lightdash/lightdash#3962]
### Description

I would like to be able to change Lightdash settings (that are currently configured in the UI) programatically. Ideally I'd be able to do this via API or terraform-style flat files. It would be great if user groups and permissions settings were also able to be configured.

### What problem does this solve?

The Terraform approach to configuration is becoming very popular because it makes life a lot easier for backend engineers. Standing up environments from scratch with all of the desired settings becomes a programmatic task rather than "click-ops". The config as code approach works better for distributed teams, and reduces the chance for human error in configuration.

**Comment by JavierLopezT** [comment]
This would be nice

**Comment by 12ian34** [comment]
@JavierLopezT I agree! Would love to hear more about your particular use case here. Do you have any specific requirements for this feature?

**Comment by yu-iskw** [comment]
The provider would be useful.
https://registry.terraform.io/providers/ubie-oss/lightdash/latest

**Comment by jpetey75** [comment]
A lot of this has been implemented recently, see docs here https://docs.lightdash.com/self-host/customize-deployment/environment-variables#organization-appearance (the last few sections towards the bottom of the page are all for initializing and customizing your instance).

If there are additional settings needed by people we can create a more specific issue.
